src/swap/handle_check_address.c

```diff
@@ -13,88 +13,94 @@
 #include "constants.h"
 #include "crc16.h"
 
-#define BASE_CHAIN             0x00
-#define MASTER_CHAIN           0xFF
-#define ADDRESS_BASE64_LENGTH  48
-#define ADDRESS_DECODED_LENGTH 36
-
-/* Set params.result to 0 on error, 1 otherwise */
-void swap_handle_check_address(check_address_parameters_t *params) {
-    PRINTF("Inside Ton swap_handle_check_address\n");
-    params->result = 0;
+#include "../jetton.h"
+#include "handle_check_address.h"
+
+/**
+ * @brief Retrieve address hash according to given derivation path.
+ */
+static bool swap_get_addr_hash_from_path(const check_address_parameters_t *const params,
+                                         uint8_t *hash,
+                                         size_t hash_len) {
+    uint8_t bip32_path_len;
+    uint32_t bip32_path[MAX_BIP32_PATH];
+    bool ret = false;
+    buffer_t cdata = {
+        .ptr = params->address_parameters,
+        .size = params->address_parameters_length,
+        .offset = 0UL,
+    };
+    pubkey_ctx_t pk_info = {
+        .subwallet_id = 698983191,
```

Comment:
I should have put a comment when writing this ... No idea where this value comes from anymore